### PR TITLE
Add contributing guidelines

### LINKED ISSUE #76: add guidelines for contributors

### PR DESCRIPTION
**Pull request by sumeet-bansal**
Closes #76.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (shravanhariharan2) [inline .github/CODEOWNERS:2]: another small nit: these are supposed to be on the same line (see [Github docs](https://docs.github.com/en/free-pro-team@latest/github/creating-cloning-and-archiving-repositories/about-code-owners)). I think that's why after rewrite you weren't getting auto-requested for reviews but I was

**Comment by sumeet-bansal** [comment]
@shravanhariharan2 ready for review, make sure to look at the Markdown files in rich text mode. Anything I forgot to cover or anything confusing in what I did cover?

Gonna let this sit for a couple days and then do a final pass with fresh eyes before merging.

File: .github/CONTRIBUTING.md
Comment (shravanhariharan2) [inline .github/CONTRIBUTING.md:2]: nit:
> ... is an open-source project, for members by members, and ...

sounds somewhat clunky. Maybe try:

> The membership portal is an open-source project made for members by members, and we welcome any contributions!"

File: .github/CONTRIBUTING.md
Comment (shravanhariharan2) [inline .github/CONTRIBUTING.md:23]: nit: "Endpoints" in rich-text mode looks smaller than the text below it (even if it's bolded). Maybe up all the header fonts by 1? Like you can just remove 1 # from every header

File: .github/CONTRIBUTING.md
Comment (shravanhariharan2) [inline .github/CONTRIBUTING.md:17]: I think you should bold the sentence "All the types in our specification... defined in the `types` folder", since I think it's super important to highlight that `types` should be the starting point for any large feature update to a newcomer

File: .github/CONTRIBUTING.md
Comment (shravanhariharan2) [inline .github/CONTRIBUTING.md:30]: nit: split this paragraph into two at "Most of these routes require...", that way its more readable and we're highlighting validation even more

File: .github/CONTRIBUTING.md
Comment (shravanhariharan2) [inline .github/CONTRIBUTING.md:37]: nit: 
> ... "business logic" of the portal, the rules of the system ..."

might be a fragment? I think replacing comma with "and" makes more sense

File: .github/CONTRIBUTING.md
Comment (shravanhariharan2) [inline .github/CONTRIBUTING.md:46]: Maybe bold the first sentence? I think generally there should be more bolding since I'm not sure how many people would read every single paragraph. Important things like validation, migrations, transactions, and others I think are worth bolding for importance

File: .github/CONTRIBUTING.md
Comment (shravanhariharan2) [inline .github/CONTRIBUTING.md:51]: I think we should add a note on documentation as well. We should mention that routes should be documented and that contributors should message us for the Postman credentials

File: README.md
Comment (shravanhariharan2) [inline README.md:2]: nit: same thing w/ for members by members

File: .github/CONTRIBUTING.md
Comment (sumeet-bansal) [reply .github/CONTRIBUTING.md:17]: The project structure section's in chronological order for changes to make. I mostly bolded sentences in the Getting Started section since people might skim over that, but if you're a newcomer you'll probably read the project structure section in its entirety.

File: .github/CONTRIBUTING.md
Comment (sumeet-bansal) [reply .github/CONTRIBUTING.md:46]: There's no bolding in the project section because first-time contributors should read all of it.

**Comment by sumeet-bansal** [comment]
Updated, but gonna solicit some feedback from Sage and Dimensions too.

**Review by shravanhariharan2 — APPROVED** [review]
Are we still waiting for anyone else's eyes on this? @michl1001 feel free to suggest any edits for additional clarity

**Comment by michl1001** [comment]
In the readme, it looks like the instructions for Macs, Linux, and Windows are all on one line, it might help to have them has separated bullets

Outside this, looks fine to me!

**Pull request merged**
